src/containers/profile/src/sagas/editCRCCase.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+
+import {
+  call,
+  put,
+  select,
+  takeEvery,
+} from '@redux-saga/core/effects';
+import { Map, get } from 'immutable';
+import { Logger } from 'lattice-utils';
+import type { Saga } from '@redux-saga/core';
+import type { WorkerResponse } from 'lattice-sagas';
+import type { SequenceAction } from 'redux-reqseq';
+
+import { submitPartialReplace } from '../../../../core/data/actions';
+import { submitPartialReplaceWorker } from '../../../../core/data/sagas';
+import { AppTypes } from '../../../../core/edm/constants';
+import { EDM, PROPERTY_FQNS_BY_TYPE_ID } from '../../../../core/redux/constants';
+import { selectEntitySetId } from '../../../../core/redux/selectors';
+import { formatNewEntityData } from '../../../../utils/data';
+import { EDIT_CRC_CASE, editCRCCase } from '../actions';
+
+const { CRC_CASE } = AppTypes;
+
+const LOG = new Logger('ProfileSagas');
+
+function* editCRCCaseWorker(action :SequenceAction) :Saga<WorkerResponse> {
+
+  const { id } = action;
+  let workerResponse :WorkerResponse;
+
+  try {
+    yield put(editCRCCase.request(id));
+    const { value } = action;
+
+    const { entityData } = value;
+    const response :Object = yield call(submitPartialReplaceWorker, submitPartialReplace({ entityData }));
+    if (response.error) throw response.error;
+
+    const crcCaseESID = yield select(selectEntitySetId(CRC_CASE));
+    const propertyFqnsByTypeId = yield select((store :Map) => store.getIn([EDM, PROPERTY_FQNS_BY_TYPE_ID]));
+
+    const crcCaseMap = get(entityData, crcCaseESID);
+    const crcCaseEKID = crcCaseMap ? Object.keys(crcCaseMap)[0] : undefined;
```

Comment:
Could `crcCaseMap` be an empty object? In that case Object.keys(crcCaseMap)[0] could be problematic

Reply:
Update: apparently reading an array index that is out of bounds returns `undefined`

Reply:
that's a good point though! i guess no change is needed?

Reply:
Yup!